File: _spike_legacy/java/com/local/dfcraftmonitor/CraftingSnapshot.java
package com.local.dfcraftmonitor;

import java.util.Collections;
import java.util.List;

public final class CraftingSnapshot {
    public final long serverNowEpochSeconds;
    public final long fetchedAtEpochMillis;
    public final List<CraftingStation> stations;

    public CraftingSnapshot(long serverNowEpochSeconds, long fetchedAtEpochMillis, List<CraftingStation> stations) {
        this.serverNowEpochSeconds = serverNowEpochSeconds;
        this.fetchedAtEpochMillis = fetchedAtEpochMillis;
        this.stations = Collections.unmodifiableList(stations);
    }
}

File: app/src/main/java/com/local/dfcraftmonitor/data/model/CraftingSnapshot.kt
package com.local.dfcraftmonitor.data.model

/**
 * 一次拉取的特勤处制造快照。stations 是不可变列表，对外封装避免下游误改。
 */
data class CraftingSnapshot(
    val serverNowEpochSeconds: Long,
    val fetchedAtEpochMillis: Long,
    val stations: List<CraftingStation>,
)
